### PR TITLE
rolls Heavy Sleeper into the new All Nighter quirk, which carries extra... baggage

### PR DESCRIPTION
**Pull request by wesoda25**
## About The Pull Request

Removes the Heavy Sleeper quirk and rolls its effects into the brand new All Nighter! On top of the effects of sleeping longer, All Nighter gives you permanent bags under your eyes as well as a permanent -3 moodlet.
## Why It's Good For The Game

I've been wanting for a way to get bags under your eyes for a while, I think it'd be a fun visual aspect that can aid with RP and whatnot. I rolled heavy sleeper into all nighter because the two are both quite similar, and I didn't feel heavy sleeper quite justified itself on it's own.

Here's what the bags look like currently on a variety of skin tones and species:
dreamseeker_Vlc4rDlHaJ

## Room for Improvement

Currently drafting for a few reasons. looking for input on all this stuff 

- [x] bag sprites could be improved
- [x] bags don't work well on moths/grayer skin tones. cest la vie?
- [x] should heavy sleeper stay? ultimately all I want is bags here, is there a better way of doing this?
- [x] all quirk elements are not properly applied and removed via VV (no need for input, i can figure this out myself)

## Changelog
:cl:
add: the All Nighter quirk!
remove: Heavy Sleeper (functionality has been rolled into All Nighter)
/:cl:

### CODE REVIEW

**Comment by wesoda25** [comment]
oh also this shamelessly rips off the implementation and design of blushing so thanks @kawoppi

**Comment by deathrobotpunch** [comment]
you can’t rip off something if it is available for everyone to use already

**Comment by wesoda25** [comment]
Just giving thanks; it is november after all

also to reiterate what i said in discord: i accidentally mislabeled the removal/del tag so plz readd that someone ty 🙏

**Comment by Higgin** [comment]
heavy sleeper was basically free, this is a good change

**Comment by Overuseddart** [comment]
A cool feature would be if caffeine (like from coffee or tea[?]) could temporarily remove the mood debuff.

File: code/datums/quirks/negative_quirks/all_nighter.dm [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	face.add_bodypart_overlay(new /datum/bodypart_overlay/simple/bags())
+	sleepy_head.update_body_parts()
+
+///removes the corresponding moodlet and visual effects
+/datum/quirk/all_nighter/remove(client/client_source)
+	quirk_holder.clear_mood_event("all_nighter", /datum/mood_event/all_nighter)
+	// this stuff is for eye bags
+	var/mob/living/carbon/human/sleepy_head = quirk_holder
+	var/obj/item/bodypart/head/face = sleepy_head.get_bodypart(BODY_ZONE_HEAD)
+	face.remove_bodypart_overlay(/datum/bodypart_overlay/simple/bags())
```
Comment (Time-Green) [inline code/datums/quirks/negative_quirks/all_nighter.dm:37]: Aside from the syntax error, this needs to be the instance

**Comment by itseasytosee** [comment]
ooh this looks cool. we should have baseball grease or war paint or something

File: code/datums/quirks/negative_quirks/all_nighter.dm [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/datum/quirk/all_nighter
+	name = "All Nighter"
+	desc = "You didn't get any sleep last night, and people can tell! You'll constantly be in a bad mood and will have a tendency to sleep longer."
+	icon = FA_ICON_BED
+	value = -4
+	mob_trait = list(TRAIT_ALL_NIGHTER, TRAIT_HEAVY_SLEEPER)
```
Comment (MrMelbert) [inline code/datums/quirks/negative_quirks/all_nighter.dm:6]: `mob_trait` does not accept lists, you have to rewrite it to allow this

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/datums/quirks/negative_quirks/all_nighter.dm (hunk 2 above)
Comment (MrMelbert) [reply code/datums/quirks/negative_quirks/all_nighter.dm:6]: ... Actually you don't *need* `TRAIT_ALL_NIGHTER`, you don't use it anywhere, so you can stick to this just giving out `TRAIT_HEAVY_SLEEPER`.

**Comment by Exester509** [comment]
Keep in mind Sleeping Carp now gives the Heavy Sleeper trait, as a joke/gimmick of #79517. I'd take that into account before touching the trait

**Comment by kawoppi** [comment]
FUCK YEAH EYEBAGS 🎉🎊🎈

**Comment by Fikou** [comment]
> Keep in mind Sleeping Carp now gives the Heavy Sleeper trait, as a joke/gimmick of #79517. I'd take that into account before touching the trait

this doesnt touch the trait, the mood and eyebags are a separate thing applied by the quirk

**Comment by wesoda25** [comment]
> A cool feature would be if caffeine (like from coffee or tea[?]) could temporarily remove the mood debuff.

Yes, I also think it'd be nice if napping did the same (maybe every second spent napping is 10 seconds without the debuff?). However I do want there to be some sort of subtle side effect from just using caffeine to circumvert sleepiness, cause personally, caffeine or not, once I hit like hour 32 I start going insane

**Comment by xzero314** [comment]
Dont remove heavy sleeper. Bad change

**Comment by SomeoneYouProbablyKnow** [comment]
Constant - 3 is a big hit to your everything. What's the cost of the quirk?
I'd be nice to get a way counteract the moodlet. Maybe actually sleeping could increase your mood only for it to drop once you wake up. Coffee is neat but make sure it doesn't just nullify the quirk.

**Comment by DATA-xPUNGED** [comment]
I like the yawn idea actually, that should be fun

**Review by DATA-xPUNGED — CHANGES_REQUESTED** [review]
One thing that's been bothering me is that the flavour of this quirk implies that it is like a one-off thing, when quirks are usually tied to the character in all rounds. It'd be neat to have this renamed to "Insomnia" or something similar

**Comment by wesoda25** [comment]
I considered something like the yawn idea but decided against it. I want this quirk to mechanically supplement RP, not replace it. Also, yes, Insomnia is an alternative, but I just thought all nighter would be more fun. Plus, insomnia would carry its own interactions with sleep that imo would go against the heavy sleeper functionality.

**Comment by carlarctg** [comment]
Quirk needs an interaction with coffee to fulfill the stereotype.

**Comment by EricZilla** [comment]
It would be cool if you got a slowly stacking negative the longer you go without sleeping instead of just perma -3 which is pretty huge ngl

File: code/datums/quirks/negative_quirks/all_nighter.dm (hunk 2 above)
Comment (wesoda25) [reply code/datums/quirks/negative_quirks/all_nighter.dm]: removed it, thanks

File: code/datums/quirks/negative_quirks/all_nighter.dm (hunk 1 above)
Comment (wesoda25) [reply code/datums/quirks/negative_quirks/all_nighter.dm]: fixed it, thanks

File: code/datums/quirks/negative_quirks/all_nighter.dm (hunk 2 above)
Comment (wesoda25) [reply code/datums/quirks/negative_quirks/all_nighter.dm]: actually, no, i still need to remove where i actually defined the trait

**Comment by wesoda25** [comment]
Made some changes, here's how things currently stand:

- 4 point negative quirk, moodlet is -3
- every second spent sleeping gives you 10 seconds without the moodlet AND eye bags
- having stimulants in your system clears the moodlet

I am a bit mixed about EricZillas suggestion and am gonna take a few days to think about it. As is, I like how non-invasive the quirk is and how it purely supplements RP, yet doesn't force it. That being said, it currently doesn't justify the -4 price tag imo, and I'm hesitant to turn it into neutral since I imagine people wouldn't like that.

**Comment by SomeoneYouProbablyKnow** [comment]
Honestly, -4 is fair since you made coffee negate it fully. That way it's similar to smoker. Also -4 also need a thing or your mood drops with the extra option of using actual sleep.
Just make sure the description explains all that.

**Comment by deathrobotpunch** [comment]
> Honestly, -4 is fair since you made coffee negate it fully. That way it's similar to smoker. Also -4 also need a thing or your mood drops with the extra option of using actual sleep.
> Just make sure the description explains all that.

no it’s still 3, you just get 4 positive quirk points for the negative quirk

**Comment by SomeoneYouProbablyKnow** [comment]
Yea which is what i'm saying. If the -3 moodlet can be negated via coffee then it puts the quirk on par with other red quirks like 4 point smoker or alcoholic.

File: code/datums/quirks/negative_quirks/all_nighter.dm (hunk 2 above)
Comment (wesoda25) [reply code/datums/quirks/negative_quirks/all_nighter.dm]: ok now its gone

**Comment by wesoda25** [comment]
calling it a day on this, i want to work on other projects. upon testing the mood debuff was negligible. that, as well as the relative ease to rid yourself of it, has made me bump it up to -5. quirk still is -4. ready for merge short of further review

**Review by Wallemations — APPROVED** [review]
Bags look fine

**Comment by coffeecoffeecoffee** [comment]
I dislike this because there is already a huge shortage of negative quirks that don't completely and totally fuck you over. Before you remove a very common negative quirk that people use to balance their characters out, maybe consider replacing it or adding in a new one?

**Pull request merged**